Микросетки на флексах для всех страниц

Redisco1 opened the pull request:
Обновил шрифты с рублем, исправил background welcome, добавил оболочку wrapper для фона и скролла.

---
:mortar_board: [Микросетки на флексах для всех страниц](https://up.htmlacademy.ru/htmlcss/33/user/1554631/tasks/8)

:boom: https://htmlacademy-htmlcss.github.io/1554631-sedona-33/7/

Redisco1 committed:
Зафлексил элементы страницы

Обновил шрифты с рублем, исправил background welcome, добавил оболочку wrapper для фона и скролла.

alicepepel commented on css/style.css:
```diff
@@ -55,34 +72,67 @@ body {
+  margin-right: 34px;
+}
+.navigation-item:last-child {
+  margin-right: 0px;
```
Отлично, что ты обнуляешь отступ у последней ссылки. Подскажу еще, как можно короче записывать такие штуки ) Два селектора в одном, с помощью not, думаю механизм прозрачен )
`.navigation-item:not(:last-child) {
margin-right: 34px;
}`

alicepepel commented on css/style.css:
```diff
@@ -55,34 +72,67 @@ body {
 }
 .navigation-link {
+  display: block;
+  padding: 20px 0px 18px;
```
Тут, скорее всего, одинаковые отступы сверху и снизу, пусть тебя не смущает подчеркивание у current ссылки. Ориентируйся на спокойное состояние, там ровно по 20px.

alicepepel commented on css/style.css:
```diff
@@ -55,34 +72,67 @@ body {
 .navigation-link-current {
 }
 .navigation-list-user {
+  justify-content: end;
```
Почему здесь ты использовал end вместо flex-end? Это не совсем ошибка, просто в учебнике и тренажерах используется flex-end, и мне интересно, откуда ты взял просто end ) Расскажу про разницу на созвоне, если это было случайно у тебя )

alicepepel commented on css/style.css:
```diff
@@ -138,31 +191,59 @@ body {
 }
 .advantages-item {
+  padding: 97px 0 87px;
   width: 400px;
```
Кстати, проглядела эту ширину. Не лучший вариант в данном случае задавать ее в пикселях. 

Почему? Нужно заранее думать о том, что размер экрана может измениться. Конечно, адаптивную верстку мы еще не проходим ) Но ты обязательно с ней столкнешься, т.к. большинство юзеров сейчас открывают сайты с мобилок. И когда ты будешь адаптировать верстку под мобилки, все размеры в пикселях тебе придется переписать, потому что 400+400+400 не поместятся полюбой )

И здесь есть более изящное решение. У нас есть  ширина родительского контейнера, максимум 1200px. А каждая item занимает четко 1/3 родителя. Так что можно написать значение в процентах, типа 33.333% ) Или что-то с помощью гридов ) И тогда item будет занимать у тебя такое пространство независимо от размера устройства, он будет сжиматься в заданном отношении.

Хотя для статичной верстки эти 400px не ошибка, просто не best practice )

alicepepel commented on css/style.css:
```diff
@@ -138,31 +191,59 @@ body {
+  margin-bottom: 0px;
+}
+.advantages-sublist .advantages-item {
+  padding: 89px 0 79px;
```
Это, кажется, можно убрать. У тебя выше и так всем .advantages-item заданы такие паддинги, должно работать. Попробуй )

alicepepel commented on css/style.css:
```diff
@@ -182,20 +263,31 @@ body {
 .hotel-search {
   width: 1200px;
```
Здесь ширину тоже можно убрать, т.к. у тебя весь wrapper по ширине 1200, а этот section лежит внутри него.

alicepepel commented on css/style.css:
```diff
@@ -204,43 +296,92 @@ body {
+  display: flex;
+  padding: 0px;
+  margin: 0 auto;
+  width: 800px;
```
Тоже можно попробовать сделать в процентах. Ну и соответственно в процентах сделать ширины инпута и кнопки. Чем больше процентов в таких элементах, тем лучше и гибче верстка ;)

alicepepel commented on css/style.css:
```diff
@@ -204,43 +296,92 @@ body {
 }
 .footer-social-item {
+  width: 18px;
+  height: 15px;
```
Вот здесь ты каждому элементу списка задаешь размеры. Подход верный - но размеры нужны ссылкам, а не элементам. Т.е. нам нужно задать область, по которой можно кликнуть, а это именно ссылка. Если мы этого не сделаем, то кликать будет некуда, т.к. в этих ссылках ничего не лежит, ни текст, ни картинки. Они 0х0 пикселей по размеру.

Нужно переписать эти селекторы на что-то типа:
`.footer-social-item:nth-child(2) a {...}`
Т.е. ссылка, которая лежит во втором элементе - такого-то размера.

Ну и конечно селектор `.footer-social-list li:nth-child(4n)` очень сложный ) Это же просто last-child, который ты уже использовал в ссылках в хедере. Т.е. последний элемент списка не должен иметь марджина, а не какой-то там х4-ый )

alicepepel commented on css/style.css:
```diff
@@ -249,15 +390,17 @@ body {
   text-align: center;
   text-transform: uppercase;
-  width: 720px;
+  width: 660px;
```
У номера точно не стоит указывать ширину. Эта ссылка шириной ровно со свой контент, т.е. номер телефона. Между ней и соцсетями/лого должно оставаться пустое пространство.

Единственное, можно задать ей max-width, чтобы она не начала залезать на другие элементы, если туда запихнут много текста. Ну вдруг )

alicepepel commented on css/style.css:
```diff
@@ -276,37 +419,68 @@ body {
   display: flex;
-  width: 500px;
-  margin: 0;
+  width: 600px;
```
А почему здесь именно 600? Не то чтобы это ошибка, просто не могу понять, откуда это значение )

alicepepel commented on css/style.css:
```diff
@@ -276,37 +419,68 @@ body {
+.breadcrumbs-item:nth-child(2) {
+  width: 5px;
+  height: 8px;
+}
```
Неа, эта маленькая стрелочка - не отдельная хлебная крошка ) Это просто декоративный элемент, который как бы подсказывает юзеру - с одной страницы мы переходим на другую. Можно предположить, что будет больше хлебных крошек - и тогда у каждой (кроме последней) будет такая стрелочка справа.

Это можно сделать либо с помощью background-image для элементов списка, либо через псевдоэлементы. В любом случае, это мелкая декоративная деталь, можно оставить ее на раздел Декоративные элементы. Просто из html нужно убрать эту стрелочку, в списке у тебя всего два элемента сейчас.

alicepepel commented on css/style.css:
```diff
@@ -276,37 +419,68 @@ body {
+  height: 8px;
+}
+.breadcrumbs-item:nth-child(3) {
+  width: 73px;
```
Ну и конечно этому элементу не нужно задавать размеры ) Его размеры определяются текстом, который в нем находится. Для первого элемента ты задаешь размеры, потому что текста внутри ссылки нет. Это просто иконка, и если ты не задашь размеры, ссылку вообще не будет видно, и на нее не кликнешь

alicepepel commented on css/style.css:
```diff
@@ -329,45 +505,60 @@ body {
 }
 .catalog-hotel {
   width: 1200px;
```
Можно убрать ширину, потому что есть wrapper )

alicepepel commented on css/style.css:
```diff
@@ -329,45 +505,60 @@ body {
+  margin-bottom: 73px;
 }
 .catalog-sorting {
-  width: 1056px;
```
Смотри, тут жесткая ширина для большого элемента. Это уже не очень хорошо, потому что не гибко )

Но главное - у тебя несколько элементов с одинаковыми отступами по краям. И они все лежат в общем родителе. Это намекает на то, что можно 1 раз сделать паддинги родителю, а не 3 раза - ширину/марджины дочерним элементам. Короче говоря, тут правильнее сделать паддинги для catalog-hotel.

alicepepel commented on css/style.css:
```diff
@@ -329,45 +505,60 @@ body {
   margin-left: auto;
   padding: 0;
-  width: 160px;
+  width: 168px;
```
А зачем ширина этому элементу? В нем ведь лежат ссылки, у которых будет своя собственная ширина, отступы и все такое. Т.е. sorting-list будет чем-то наполнен, и это будет его ширину создавать.

alicepepel commented on css/style.css:
```diff
@@ -418,31 +610,51 @@ body {
   text-align: center;
 }
 .pagination-list {
+  width: 576px;
```
Та же история, что и с sorting-lis, зачем ему ширина, если внутри лежит содержимое? Можно было бы сделать max-width, но тут и ограничивать ничего не надо - кнопок уже ограниченное количество, между первыми и последней - троеточие )

alicepepel commented on css/style.css:
```diff
@@ -418,31 +610,51 @@ body {
+  flex-wrap: wrap;
 }
-.pagination-item {
+.pagination-item:not(:nth-child(9n)) {
```
Такие селекторы используются в каких-то супер-сложных случаях, когда у тебя дофига элементов, и нужно именно каждому девятому что-то назначить ) Скорее всего, тут `:not(:nth-child(9n))` можно заменить на `:not(:last-child)`.

alicepepel commented on css/style.css:
```diff
@@ -418,31 +610,51 @@ body {
   margin-right: 9px;
 }
 .pagination-link {
+  width: 56px;
```
Кстати, а зачем задавать размеры и item, и link? Не проще ли чему-то одному, ведь по сути это дублирование ) 

И еще нюанс - вот ты задаешь размеры 55х55, например. А если страниц станет много-много, и понадобится вписать в пагинацию 3-значное число? Влезет ли оно в эти размеры? Обязательно такие штуки проверяй )

alicepepel commented on css/style.css:
```diff
@@ -418,31 +610,51 @@ body {
 }
 .pagination-selectlist {
+  display: flex;
+  width: 386px;
```
Ну, это тоже не стоит ограничивать по ширине и высоте (ну хотя бы не одновременно)). Потому что внутри есть текст "Гостиниц на странице:", который всегда может стать больше. Да и в принципе, это ограничение точно здесь нужно?

alicepepel commented on css/style.css:
```diff
@@ -418,31 +610,51 @@ body {
+  font-size: 18px;
+  line-height: 23px;
+  display: inline-block;
+  margin: 16px 51px 16px 0px;
```
Здесь по сути нужен только один марджин - правый, это отступ от селекта. Левый обнулять не нужно, а верхний и нижний - они от чего?

Redisco1 committed:
Исправил ошибки в микросетках на флексах

Убрал ограничивающую  ширину элементов Хедера,  сортировки, пагинации. В преимуществах заменил единицы измерения ширины с px на %.  Исправил отступы у ссылок, добавил размер ссылкам, для кликабельной области. Кнопкам button  добавил параметр - cursor: pointer;. Немного стилизовал select (фон, рамку, размер). Сделал сетку карточек товаров на гридах.
Исправил ссылки в хлебных крошках  , переназначил стили webkit.

Redisco1 committed:
Проработал все width

Исправил всю ширину, возможно, где то пропустил.
